enclave/sgx/start.S

```diff
@@ -13,28 +13,21 @@
 .type _start, @function
 _start:
 .cfi_startproc
+    /* Save the registers passed by EENTER. */
+    pushq %rax
+    pushq %rbx
+    pushq %rcx
+    pushq %rdi
+    pushq %rsi
+    /* Dynamically link liboecore to liboeenclave (install upcalls). */
+    call oe_link_enclave
```

Comment:
>call oe_link_enclave [](start = 4, length = 20)

Can we move the call to oe_enter, so that we can potentially do away with those register save/restores?